pkg/registry/rbac/clusterrole/policybased/storage.go

```diff
@@ -68,10 +80,21 @@ func (s *Storage) Update(ctx genericapirequest.Context, name string, obj rest.Up
 
 		rules := clusterRole.Rules
 		if err := rbacregistryvalidation.ConfirmNoEscalation(ctx, s.ruleResolver, rules); err != nil {
-			return nil, errors.NewForbidden(groupResource, clusterRole.Name, err)
+			return nil, apierrors.NewForbidden(groupResource, clusterRole.Name, err)
 		}
+		// to change the aggregation rule, since it can gather anything and prevent tightening, requires * on *.*
+		if hasAggregationRule(clusterRole) || hasAggregationRule(oldClusterRole) {
+			if err := rbacregistryvalidation.ConfirmNoEscalation(ctx, s.ruleResolver, fullAuthority); err != nil {
+				return nil, apierrors.NewForbidden(groupResource, clusterRole.Name, errors.New("must have cluster-admin privileges to use the aggregationRule"))
+			}
+		}
```

Comment:
if we have aggregation rules and validate fullAuthority, we don't need the detailed covers check on the role rules

Reply:
> if we have aggregation rules and validate fullAuthority, we don't need the detailed covers check on the role rules

I'd really rather put it in an else branch branch unless you feel strongly.  It's not like these are hit often and the unconditional check is very easy to read and reason about.  Makes me feel safe.